### ISSUE TITLE
Is it possible to satisfy the no-duplicate-imports rule with type imports?

### ISSUE THREAD

**Issue by garth** [blakeembrey/tslint-config-standard#41]
Is it possible to import from both a node_module and type in a single import statement?

```js
import * as React from 'react'; // import from module
import { StatelessComponent } from 'react'; // import from type
```

**Comment by blakeembrey** [comment]
Should be. Try `import * as React, { StatelessComponent } from React`.

**Comment by garth** [comment]
I don't think that is valid syntax.

**Comment by blakeembrey** [comment]
You're right, sorry. I'm not sure then. I'd just use destructing like normal: 

```js
import * as React from 'react'
const { StatelessComponent } = React
```

**Comment by blakeembrey** [comment]
If you'd like, feel free to verify the behaviour is the same as with StandardJS. They're using it from https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-duplicates.md, so there may be a deviation in TSLint if your code was valid with ESLint.

**Comment by garth** [comment]
I think the rule is valid for is, just not sure it should be applied to ts. I have disabled the rule in local config, but I would be interested to see if there is a better solution.

**Comment by blakeembrey** [comment]
Both imports are identical in JS vs TS, there's no _real_ difference. If what you mean is about `StatelessComponent` as a type, you can import from a namespace but it sadly gets ugly. Try this:

```ts
namespace React {
    export interface StatelessComponent { }
}

import StatelessComponent = React.StatelessComponent
```

This would be improved 10x if TypeScript supported destructing in this case, which i think is pretty important for property parity with JS nowadays (e.g. `import { Component } = require('react')` should be allowed by TypeScript too, but the only allow `import React = require('react')`).

**Comment by blakeembrey** [comment]
Ah, sorry, I just saw you were talking about type imports in the title 😄 I regularly overlook the title and just read the body, sorry for the confusion and oversight.

**Comment by Hotell** [comment]
@garth what you should do in case of importing react stuff is following:

**tsonfig:**

```json
{
  "compileOptions": {
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true
  }
}
```

now you can:

```ts
import React, { StatelessComponent } from 'react';
```

PRO TIP: instead of importing super long/verbose `StatelessComponent` use `SFC` type alias

PRO TIP 2: We've gone even further and import only what the module needs. So for transforming JSX to function calls we import only `createElement`

```ts
// now you got both runtime function and typescript type in one import
import {createElement, SFC} from 'react'
``` 

**tsonfig:**

```json
{
  "compileOptions": {
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "jsx": "react",
    "jsxFactory": "createElement",
  }
}
```
